### ISSUE TITLE
Is there a new version of this for Teensy 4.0? I'm trying to port it.

### ISSUE THREAD

**Issue by pinaa** [kripthor/teensy31-micloc-java#2]
I am using the Teensy 4.0 for this project and the  kripthor /
teensy31-micloc  that feeds this. It compiles and seems to work except Java code seems to constantly RESync. This is a great project and I would love to get this working, but I just don't have the expertise to make any real progress without some guidance.

**Comment by thedead** [comment]
Did you every get a chance to do this?

**Comment by pinaa** [comment]
Yes, I did. I archived them in my local repos as teensy40-micloc-java.zip, teensy40micloc.zip.
In TeensySimultaneousADC.ino, I changed byte to int to get 12-bit v 8-bit resolution and later used 10-bit. Implemented flow-control. My comments indicate that I used library.json version 9.1.0 v 8.0.38 for arduino-1.8.13.
I also dropped the fourth channel because I only wanted to use 3 mics.
My setup() does,
`adc->adc0->setResolution(10);                                   			// set bits of resolution
  adc->adc0->setConversionSpeed(ADC_CONVERSION_SPEED::HIGH_SPEED); 			// change the conversion speed
  adc->adc0->setSamplingSpeed(ADC_SAMPLING_SPEED::HIGH_VERY_HIGH_SPEED);// change the sampling speed
	adc->adc1->setConversionSpeed(ADC_CONVERSION_SPEED::HIGH_SPEED);
  adc->adc1->setSamplingSpeed(ADC_SAMPLING_SPEED::HIGH_VERY_HIGH_SPEED);`
The loop() has sample loop to read data into the buffer with,
` //TAKE THE READINGS
    buffer1[i]=value1 = adc->adc0->analogRead(A2);
    buffer2[i]=value2 = adc->adc1->analogRead(A3);
    buffer3[i]=value3 = adc->analogRead(A4);    // Use the less busy ADC
`
The code worked when I hosted it on my Ubuntu desktop (Intel® Core™ i7-5820K CPU @ 3.30GHz × 12), but I wanted to automate it so I could get emails with incident reports and summaries, but I didn't have the bandwidth. I'm retired and have a lot of other projects, so I didn't pursue this after May 2021. I bought a Lo-Ra module I was planning to use with this to handle RF communications, but I never followed through after doing some preliminary testing of the transmission modules. My plan was to use the Teensy to do the sound triangulation and an ESP32 to run the Java code and communicate to my Odroid server via Lo-Ra which could then communicate with any client over the Internet. I borrowed a Raspberry Pi to evaluate that platform as an alternative server, but it's way too expensive due to the supply shortage (they're about 4-10x above the list price, but I'm sure they'll be reasonably priced in late 2024, assuming inflation and US chip incentives don't create a strong negative impact on this British controller and its Chinese manufacturers).

**Comment by thedead** [comment]
Thanks for the additional info!  Did you ever publish those repos?  

I have 4 sound detectors running on an Arduino, but I would like to improve the accuracy by using microphones with ADC as opposed to digital.

**Comment by pinaa** [comment]
No, I didn't publish it because I wanted to ask for your review.
And since I stripped a lot of comments and added my comments because it was
much easier for me to follow what I was doing with this approach, I didn't
want you or anyone else to miss their due credit. Also, I didn't credit the
help from the group which I thought would be a good idea, though perhaps
not in the code to avoid bloat, but certainly where appropriate.

Perhaps you can comment on using LoRa 1276 as a packet forwarder. In my
case, I was planning to use LoRa to communicate between the sensor unit and
the master hub which would be tied to the LAN so I could use the data
locally and forward it over the Internet if needed.

Thanks & Regards,
Art
